cmplibrary/src/main/java/com/sourcepoint/cmplibrary/campaign/CampaignManager.kt

```diff
@@ -650,4 +652,45 @@ private class CampaignManagerImpl(
             ccpaCS = ccpaConsentStatus
         )
     }
+
+    override val isGdprExpired: Boolean
+        get() {
+
+            val formatter = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.getDefault())
+            val gdprExpirationDate = dataStorage.gdprExpirationDate?.let { formatter.parse(it) } ?: return false
```

Comment:
I think `SimpleDateFormat.parse()` might throw `ParseException`, we should keep that in mind, maybe even wrap up this call in `try/catch` and return `false` if the exception being thrown

BUT, that might give us a false positive result, when we will not be able to get the issue...

Reply:
Exactly if the format changes and we don’t get an exception we can have a false positive

Reply:
@carmelo-iriti so what's the solution? if `formatter.parse(it)` throws an exception, the gdpr consent will never expire?

Reply:
There are 2 cases that we can have a crash:
- the value to parse is null but we are already have a null check
- the value from the backend changes and we fallback to the graceful degradation.
If something changes in the backend, we'll know from our Datadog metrics before next year when the consent will expire